### DISCUSSION TITLE
StrongHandle did not register ReflectSerialize

### DISCUSSION THREAD

**Discussion by Qinka** [bevyengine/bevy#14281] [Q&A]
My code:
```rust

fn dump_scene(
  world: &mut World,
) {

  let mut scene_world = World::new();

  let type_registry = world.resource::<AppTypeRegistry>().clone();
  scene_world.insert_resource(type_registry);

  let shape = Mesh::from(Rectangle::new(100., 100.));
  let color = ColorMaterial::from(Color::srgb(1., 0., 0.));
  let mesh_handle = scene_meshes.add(shape);
  let material_handle = scene_materials.add(color);

  scene_world.spawn( (
    ball::BallBundle::new(20., 20.),
    mesh_handle,
    material_handle,
  ));

  let scene = DynamicScene::from_world(&scene_world);
  let type_registry = world.resource::<AppTypeRegistry>();
  let type_registry = type_registry.read();
  let serialized_scene = scene.serialize(&type_registry).unwrap();

  warn!("SCN: {}", serialized_scene);
}
```
But I get error `called `Result::unwrap()` on an `Err` value: Message("Type 'std::sync::Arc<bevy_asset::handle::StrongHandle>' did not register ReflectSerialize")`

What should I do, if I want to load an entity in the scene with `bevy::sprite::Mesh2dHandle` to add a mesh to entity?

**Comment by Syvies** [comment]
I get the exact same error when trying to serialize a SceneBundle.
I would guess that some of the components spawned by one or more bundles can't be serialized. Would be great if Bevy could ignore un-serializable components instead of panicking....

**Reply by nuzzles** [reply]
This is possible, you can `#[serde(skip)]` (for Serialize) or `#[reflect(ignore)]`.

**Comment by yrns** [comment]
> What should I do, if I want to load an entity in the scene with `bevy::sprite::Mesh2dHandle` to add a mesh to entity?

Unfortunately, Bevy does not serialize asset handles. Even if it did, you would need an asset server to deserialize them. I believe this is being addressed by the scene rework described [here](https://github.com/bevyengine/bevy/discussions/14437).

In your case you could filter out the components containing handles with the [DynamicSceneBuilder](https://docs.rs/bevy/latest/bevy/prelude/struct.DynamicSceneBuilder.html), make a custom component that serializes to an asset path, and then generate the handles with some startup system from your component. You can also forego Bevy scenes and write a custom asset loader that handles a custom serializable asset.
